src/metrics/spatial.jl

```diff
@@ -122,3 +125,105 @@ function summarize(
 )::YAXArray where {D,T,N,A}
     return summarize(data[timesteps=timesteps], alongs_axis, metric)
 end
+
+"""
+    ensemble_loc_difference(outcome::YAXArray{T,3}, scens::DataFrame; agg_metric::Union{Function,AbstractFloat}=median, diff_target=:guided, conf::Float64=0.95, rand_seed=1234)::YAXArray where {T}
+
+Mean bootstrapped difference (counterfactual - target) between some outcome aggregated for
+each location.
+
+# Arguments
+- `outcome` : Metric outcome with dimensions (:timesteps, :locations, :scenarios).
+- `scens` : Scenarios DataFrame.
+- `agg_metric` : Metric used to aggregate scenarios when comparing between counterfactual and
+target. If it is an `AbstractFloat` between 0 and 1, it uses the `bs_metric`-th quantile.
+Defaults to `median`.
+- `diff_target` : Target group of scenarios to compare with. Valid options are `:guided` and
+`:unguided`. Defaults to `:guided`
+- `conf` : Percentile used for the confidence interval. Defaults to 0.95.
+- `rand_seed` : Seed used for random number generator. Defaults to 1234.
+
+# Example
+```
+# Load domain
+dom = ADRIA.load_domain(path_to_domain)
+
+# Create scenarios
+num_scens = 2^6
+scens = ADRIA.sample(dom, num_scens)
+
+# Run model
+rs = ADRIA.run_scenarios(dom, scens, "45")
+
+# Calculate difference to the counterfactual for given metric
+_relative_cover = metrics.relative_cover(rs)
+
+# Compute difference between guided and counterfactual using the 0.6-th quantile
+gd_res = metrics.ensemble_loc_difference(r_cover, scens; agg_metric=0.6)
+
+# Compute difference between unguided and counterfactual using the median
+ug_res = metrics.cf_difference_loc(r_cover, scens; diff_target=:unguided)
+
+# Plot maps of difference to the counterfactual
+ADRIA.viz.diff_map(rs, gd_res[2, :])
+ADRIA.viz.diff_map(rs, ug_res[2, :])
+```
+
+# Returns
+Vector with bootstrapped difference (counterfactual - guided) for each location.
+"""
+function ensemble_loc_difference(
+    outcome::YAXArray{T,3},
+    scens::DataFrame;
+    agg_metric::Union{Function,AbstractFloat}=median,
+    diff_target=:guided,
+    conf::Float64=0.95,
+    rand_seed=1234
```

Comment:
Minor point - could we use a global random seed instead?

```suggestion
    rng::AbstractRNG=Random.GLOBAL_RNG
```

Later on...

`shuffle(rng, 1:n_cf_outcomes)[1:min_n_outcomes]`

Reply:
Does that mean that the user shouldn't be allowed to set the seed and I should also remove this `Random.seed!(rand_seed)` ?

Reply:
Yes, you should remove the `Random.seed!` inside the function.

It means the user can set it outside the function.

This way the state of the random number generator does not "leak" outside the scope of the function.
Currently, the function "resets" the seed state, which may affect other functions.

Best practice is to set a RNG and pass that to each function if necessary

```julia
rng = Random.Xoshiro(123)  # use a different RNG algorithm
some_function(args...; rng=rng)
another_function(args...; rng=rng)
```

As an aside: the current exception in ADRIA is the scenario running, where I made the decision that we want the outcome to be tied to the scenario factors. But it's implemented in a suboptimal way... another thing to revisit.

Reply:
Makes sense, thanks for the explanation.